ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,124 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+import re
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
 
-class RedHatAuthConfig(object):
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
```

Comment:
1. Don't you need to make something equivalent to `statestore.backup_state('authconfig', 'sssd', True)`, for instance `statestore.backup_state('authselect', 'sssd', True)` ?

2. The message states that it is not supported but allows the installer to go on. It is somehow puzzling.

3. If I run ipa-client-install --no-sssd, I can see this line printed in the output. As a good citizen, I then run `ipa-advise config-fedora-authconfig` which provides the following script:
```
#!/bin/sh
# ----------------------------------------------------------------------
# Authconfig instructions for configuring Fedora 18/19 client with IPA
# server without use of SSSD.
# ----------------------------------------------------------------------
# Run the following command as a root:
/sbin/authconfig --enableldap --ldapserver=vm-171-220.abc.idm.lab.eng.brq.redhat.com --enablerfc2307bis --enablekrb5
```
and trying to run the script fails with:
```
Running authconfig compatibility tool.

IMPORTANT: authconfig is replaced by authselect, please update your scripts.
See Fedora 28 Change Page: https://fedoraproject.org/wiki/Changes/AuthselectAsDefault
See man authselect-migration(7) to help you with migration to authselect

Error: Please, provide --updateall option.
```